lib/brain.js

```diff
@@ -2564,8 +2540,8 @@ Brain.prototype._startAddressScan = function _startAddressScan () {
 
   const frames = []
   const saveFailedScans = this.shouldSaveQrAttempts ?
-    readFailedFrames(photo => frames.push(photo)) :
-    null
+    photo => frames.push(photo) :
+    _.noop
```

Comment:
There's no limit to the number of frames here, and `lib/scanner-node.js:capture()` calls it on every single failed frame (at most 1 frame/s), which will surely fail to upload in `Brain.failedQRScans()`.